File: android project/app/src/main/java/com/example/andriod/otplessdemoandroid/MainActivity.java
package com.example.andriod.otplessdemoandroid;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        parseOtplessIntent(getIntent());
        setContentView(R.layout.activity_main);
        Button button = (Button) findViewById(R.id.button_open_auth_link);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://gauge.authlink.me"));
                startActivity(browserIntent);
            }
        });
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        // Get the Intent that started this activity
        parseOtplessIntent(intent);
    }

    private void parseOtplessIntent(Intent intent){
        // Checking intent contains value and otpless intent only
        if (intent != null && intent.getData() != null && intent.getData().getHost().contains("otpless")){
            Uri redirectUri = intent.getData();
            Log.d("url",redirectUri.toString());
            String urlString = redirectUri.toString();
            String waId = redirectUri.getQueryParameter("waId");
            Log.d("waId", waId);
        }
    }


}

File: Otpless Demo/app/src/main/java/com/example/andriod/otplesskotlindemo/MainActivity.kt
package com.example.andriod.otplesskotlindemo

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        parseOtplessIntent(intent)
        setContentView(R.layout.activity_main)
        var button = findViewById(R.id.button_open_auth_link) as Button
        button.setOnClickListener {
            val urlIntent = Intent(
                Intent.ACTION_VIEW,
                Uri.parse("https://gauge.authlink.me")
            )
            startActivity(urlIntent)
        }
    }
    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        parseOtplessIntent(intent)
    }

    private fun parseOtplessIntent(intent: Intent?) {
        if (intent != null && intent.data != null && intent.data!!.host!!.contains("otpless")) {
            val redirectUri = intent.data
            println(redirectUri)
            val waId = redirectUri!!.getQueryParameter("waId")
            println(waId)
            if (waId != null) {
                Toast.makeText(this, waId, Toast.LENGTH_SHORT).show()
            }
        }
    }
}